tree/dataframe/inc/ROOT/RDF/RDatasetSpec.hxx

```diff
@@ -0,0 +1,83 @@
+// Author: Vincenzo Eduardo Padulano CERN/UPV, Ivan Kabadzhov CERN  06/2022
+
+/*************************************************************************
+ * Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_RDF_RDATASETSPEC
+#define ROOT_RDF_RDATASETSPEC
+
+#include <string>
+#include <vector>
+#include <limits>
+#include <utility>
+
+#include <RtypesCore.h>               // Long64_t
+#include <ROOT/InternalTreeUtils.hxx> // ROOT::Internal::TreeUtils::RFriendInfo
+
+namespace ROOT {
+
+namespace Detail {
+namespace RDF {
+class RLoopManager;
+} // namespace RDF
+} // namespace Detail
+
+namespace RDF {
+
+namespace Experimental {
+
+class RDatasetSpec {
+
+   friend class ROOT::Detail::RDF::RLoopManager;
+
+   struct REntryRange {
```

Comment:
This breaks Python compatibility:

```python
>>> ds = ROOT.RDF.Experimental.RDatasetSpec("mytree", "myfile.root")
input_line_43:19:59: error: 'REntryRange' is a private member of 'ROOT::RDF::Experimental::RDatasetSpec'
            (const ROOT::RDF::Experimental::RDatasetSpec::REntryRange&)*(const ROOT::RDF::Experimental::RDatasetSpec::REntryRange*)args[2]);
                                                          ^
/home/vpadulan/programs/rootproject/rootbuild/distrdf-fix-friends-distrdf-debug/include/ROOT/RDF/RDatasetSpec.hxx:38:11: note: implicitly declared private here
   struct REntryRange {
```

This is also a reminder that we should add Python tests before merging

Reply:
Okay, I will prepare python tests. I consider to rewrite the C++ tests in python, as they should cover all cases that I am aware of.

Buuut, then my question is. What to do with this struct. Should I simply move it out of the class? (And it will be in the ns ROOT::RDF::Experimental)

Reply:
Okay *major* TODO left for me! Right python tests!